sdk/openai/Azure.AI.OpenAI/src/Custom/EmbeddingsOptions.cs

```diff
@@ -4,22 +4,42 @@
 #nullable disable
 
 using System;
+using System.Collections.Generic;
 using Azure.Core;
 
 namespace Azure.AI.OpenAI
 {
     /// <summary> Schema to create a prompt completion from a deployment. </summary>
     public partial class EmbeddingsOptions
     {
-        internal string InternalNonAzureModelName { get; set; }
+        /// <summary>
+        /// Gets or sets the deployment name to use for an embeddings request.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// When making a request against Azure OpenAI, this should be the customizable name of the "model deployment"
+        /// (example: my-gpt4-deployment) and not the name of the model itself (example: gpt-4).
+        /// </para>
+        /// <para>
+        /// When using non-Azure OpenAI, this corresponds to "model" in the request options and should use the
+        /// appropriate name of the model (example: gpt-4).
+        /// </para>
+        /// </remarks>
+        [CodeGenMember("InternalNonAzureModelName")]
+        public string DeploymentName { get; set; }
 
-        /// <inheritdoc cref="EmbeddingsOptions.EmbeddingsOptions(System.Collections.Generic.IEnumerable{string})"/>
-        public EmbeddingsOptions(string input)
-            : this(new string[] { input })
-        {
-        }
+        /// <summary>
+        /// Input texts to get embeddings for, encoded as a an array of strings.
+        /// Each input must not exceed 2048 tokens in length.
+        ///
+        /// Unless you are embedding code, we suggest replacing newlines (\n) in your input with a single space,
+        /// as we have observed inferior results when newlines are present.
+        /// </summary>
+        public IList<string> Input { get; set; } = new ChangeTrackingList<string>();
```

Comment:
Is the purpose of this customization to instantiate a non-empty collection for the case of the parameter-less constructor? If so, would it be better to let the property be auto-generated (without the initialization code) and then initialize it in the parameter-less constructor? I think that would make it more obvious why the code is customized. Alternatively, we can add a note to this custom property explaining why it was customized.

Reply:
I believe the reason that this property (and many others) are customized is to add richer XML comments than what the TypeSpec emits. I'd love to eliminate most of that if we can (Completions and Chat Completions are especially big offenders) -- it's likely tackled best in its own, dedicated change.